framework/src/loops/ComputeMortarFunctor.C

```diff
@@ -112,7 +118,10 @@ ComputeMortarFunctor::operator()(const Moose::ComputeType compute_type)
       case Moose::ComputeType::ResidualAndJacobian:
       {
         for (auto * const mc : _mortar_constraints)
+        {
+          mc->setNormals();
```

Comment:
Mental note: It appears we would have to compute tangents here as well, which will be needed for any kind of frictional modeling/enforcement.